core/src/main/scala/org/apache/spark/deploy/security/HBaseDelegationTokenProvider.scala

```diff
@@ -39,23 +41,46 @@ private[security] class HBaseDelegationTokenProvider
       creds: Credentials): Option[Long] = {
     try {
       val mirror = universe.runtimeMirror(Utils.getContextOrSparkClassLoader)
-      val obtainToken = mirror.classLoader.
-        loadClass("org.apache.hadoop.hbase.security.token.TokenUtil").
-        getMethod("obtainToken", classOf[Configuration])
-
-      logDebug("Attempting to fetch HBase security token.")
-      val token = obtainToken.invoke(null, hbaseConf(hadoopConf))
-        .asInstanceOf[Token[_ <: TokenIdentifier]]
-      logInfo(s"Get token from HBase: ${token.toString}")
-      creds.addToken(token.getService, token)
-    } catch {
-      case NonFatal(e) =>
-        logWarning(s"Failed to get token from service $serviceName", e)
+      // Token<AuthenticationTokenIdentifier> obtainToken(Configuration conf) is a deprecated
+      // method and in Hbase 2.1.1 the method is already removed. there is one more public
```

Comment:
Could you be more specific? Is this removed at 2.1.1?

Reply:
shall we mention as 2.x, they removed it as part of Hbase 2.0.0 major version upgrade

Reply:
https://issues.apache.org/jira/browse/HBASE-14713_ - JIRA for reference

Reply:
updated the version. thnks .